Tasks/PackerBuildV0/src/operations/packerBuild.ts

```diff
@@ -26,23 +26,44 @@ export async function run(packerHost: packerHost): Promise<any> {
     command.arg(packerHost.getTemplateFileProvider().getTemplateFileLocation(packerHost));
 
     console.log(tl.loc("ExecutingPackerBuild"));
-    var outputVariablesParser: definitions.IOutputParser = new OutputVariablesParser([constants.PackerLogTokenImageUri, constants.PackerLogTokenStorageLocation]);
+    var taskParameters = packerHost.getTaskParameters();
+    var outputVariablesParser: definitions.IOutputParser;
+    if(!taskParameters.isManagedImage){
+        outputVariablesParser = new OutputVariablesParser([constants.PackerLogTokenImageUri, constants.PackerLogTokenStorageLocation]);
+    } else {
+        outputVariablesParser = new OutputVariablesParser([constants.PackerLogTokenManagedImageName, constants.PackerLogTokenManagedResourceGroupName, constants.PackerLogTokenManagedImageLocation]);
+    }
+
     await packerHost.execTool(command, outputVariablesParser);
 
     // set output task variables
     setOutputVariables(packerHost, outputVariablesParser.getExtractedOutputs());
 }
 
 function setOutputVariables(packerHost: packerHost, outputs: Map<string, string>): void {
-    var imageUri = outputs.get(constants.PackerLogTokenImageUri);
-    var taskParameters = packerHost.getTaskParameters();
+   var taskParameters = packerHost.getTaskParameters();
+   var imageUri;
+   if(!taskParameters.isManagedImage) {
```

Comment:
there are too many similar if conditions in this function - see if you can refactor it to make it simpler

Reply:
@bishal-pdMSFT ave refactored the function, please let me know if you are fine with it.